[TT-17415] Update documentation for master (#2487)
TykTechnologies/tyk-docs: snippets/gateway-config.mdx

Type: `int`<br />

Set this to the number of seconds that Tyk uses to flush content from the proxied upstream connection to the open downstream connection.

> This option needed be set for streaming protocols like Server Side Events, or gRPC streaming.

Reviewer: ```suggestion
This option needs to be set for streaming protocols like gRPC streaming.
```
The documentation previously recommended setting http_server_options.flush_interval for SSE APIs. However, the Gateway automatically handles immediate flushing for text/event-stream responses (hardcoded to -1), so configuring this is unnecessary and can lead to other issues.

Refer this [ticket](https://tyktech.atlassian.net/browse/DX-2390)

Reviewer 2: Addressed in https://github.com/TykTechnologies/tyk/pull/8512

### policies.policy_source
ENV: <b>TYK_GW_POLICIES_POLICYSOURCE</b><br />

> Type: `PolicySource`<br />

Reviewer: Can we please stick to standard Golang types? No custom types allowed, https://tyk.io/docs/tyk-oss-gateway/configuration#environment-variable-type-mapping

Reviewer 2: @sharadregoti addressed in https://github.com/TykTechnologies/tyk-config-info-generator/pull/21

If you set `disable_regexp_cache` to `false`, you can use this setting to limit how long the regular expression cache is kept for in seconds.
The default is 60 seconds. This must be a positive value. If you set to 0 this uses the default value.

> ### regexp_cache_max_entries

Reviewer 3: Where did these settings come from? They are really not intuitively named or described.

Reviewer 2: @andyo-tyk looks like this was added in TT-17049

### log_format
ENV: <b>TYK_GW_LOGFORMAT</b><br />

> Type: `logger.Format`<br />

Reviewer: Same

Reviewer 2: Addressed in https://github.com/TykTechnologies/tyk-config-info-generator/pull/21

> LogFormat configures the output format of the logs.

Reviewer 3: ```suggestion
Configures the output format used for application logs.
```

effective since, in those cases, you're either recording everything or nothing, and there are no
intermediary decisions to consider. The default value for this option is false.

### opentelemetry.traces.mcp
ENV: <b>TYK_GW_OPENTELEMETRY_TRACES_MCPTRACECONTEXT</b><br />
Type: `MCPTraceContextConfig`<br />

> MCPTraceContext configures where Tyk reads an MCP request's W3C trace

Reviewer 3: ```suggestion
Configure from where Tyk should read an MCP request's W3C
```

> context from (ordered, first-match-wins). Omitted ⇒ the canonical

Reviewer 3: ```suggestion
 trace context (ordered, first-match-wins). If not set, the canonical
```

> [{header}, {body, path: params._meta}] default.

Reviewer 3: ```suggestion
default [{header}, {body, path: params._meta}] will be used.
```
